### PR TITLE
Make PSSA look for a file in the workspace root by default

### PR DESCRIPTION
**Pull request by rjmholt**
## PR Summary

Fixes https://github.com/PowerShell/vscode-powershell/issues/2190.

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ export function load(): ISettings {
 
     const defaultScriptAnalysisSettings: IScriptAnalysisSettings = {
         enable: true,
-        settingsPath: "",
+        settingsPath: "PSScriptAnalyzerSettings.psd1",
```
Comment (TylerLeonhardt) [inline src/settings.ts]: I think we can also set this in the `package.json`

**Comment by bergmeister** [comment]
@TylerLeonhardt @rjmholt 
Have you guys actually tested this change?
Because I don't think this works at all because the logic a few layers underneath is something along the lines:
If the setting is not defined, then pass the default set of PSSA rules as settings as defined in PSES, if it is defined, pass that file path. There is no checking that the file actually exists and therefore with this default, you've just turned off PSSA warnings for 95% of the users, which are the ones not using a PSSA settings file... PSSA itself throws an error if the settings file does not exist but I mentioned previously that for some reason, PSES ignores or does not propagate PSSA errors.

**Comment by TylerLeonhardt** [comment]
All of that logic you speak of was added in this PR:

https://github.com/PowerShell/PowerShellEditorServices/pull/1179

**Comment by rjmholt** [comment]
@bergmeister are you saying that you’re experiencing the behaviour you’re describing?

**Comment by bergmeister** [comment]
First of all, mea culpa, I did not see the other recent PSES change, thanks for pointing me to it. But at the same point I also pulled the latest changes from PSES and vscode-powershell from master into my forks and yes, I was experiencing the behaviour of PSSA warnings not showing up at all.

**Comment by rjmholt** [comment]
I’m assuming you don’t have a `PSScriptAnalyzerSettings.psd1` defined in the root of the workspace where you’re seeing this? (Since that would result in behaviour like this — PSSA doesn’t have much in the way of error reporting on settings)

**Comment by bergmeister** [comment]
Yes, I don't have a settings file at the root (deliberately, to mirror what most people work with every day).
You cannot just disregard the most common use case and put the blame on PSSA, this is a big breaking change.
A few months ago, I tried to do a prototype to check if the settings file exists by writing some typescript code in the vscode-powershell repo to check if a file named `PSScriptAnalyzerSettings.psd1` exists (which did work well and which I suggest to be implemented) but the reason why I did not proceed further is because I couldn't change the code to use that discovered settings path (because of the global nature of the setting, I couldn't just pass the value to something on a lower level or change the setting value behind the scenes)

**Comment by TylerLeonhardt** [comment]
@bergmeister it's a bug. Please understand we wouldn't break the customer like that. That would be silly.

**Comment by rjmholt** [comment]
Migrating this discussion to https://github.com/PowerShell/vscode-powershell/issues/2489

**Pull request merged**